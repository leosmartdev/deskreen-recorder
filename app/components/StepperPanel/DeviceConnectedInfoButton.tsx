import React from 'react';
import { Row, Col } from 'react-flexbox-grid';
import { Icon, Text, Button, Popover, Tooltip } from '@blueprintjs/core';
import { useTranslation } from 'react-i18next';
import { TFunction } from 'i18next';
import DeviceInfoCallout from '../DeviceInfoCallout';
import isWithReactRevealAnimations from '../../utils/isWithReactRevealAnimations';

interface DeviceConnectedInfoButtonProps {
  device: Device;
  onDisconnect: () => void;
}

const getDeviceConnectedPopoverContent = (
  t: TFunction,
  pendingConnectionDevice: Device,
  handleDisconnect: () => void
) => {
  const disconnectButtonText = t('Disconnect');

  return (
    <Row>
      <div style={{ padding: '20px', borderRadius: '100px' }}>
        <Row style={{ margin: '0 px 10px 10px 10px' }}>
          <DeviceInfoCallout
            deviceType={pendingConnectionDevice?.deviceType}
            deviceIP={pendingConnectionDevice?.deviceIP}
            deviceOS={pendingConnectionDevice?.deviceOS}
            sharingSessionID={pendingConnectionDevice?.sharingSessionID}
            deviceBrowser={pendingConnectionDevice?.deviceBrowser}
          />
        </Row>
        <Row>
          <Col xs={12}>
            <Button
              intent="danger"
              icon="disable"
              onClick={() => {
                handleDisconnect();
              }}
              style={{ width: '100%', borderRadius: '100px' }}
            >
              {disconnectButtonText}
            </Button>
          </Col>
        </Row>
      </div>
    </Row>
  );
};

export default function DeviceConnectedInfoButton(
  props: DeviceConnectedInfoButtonProps
) {
  const { t } = useTranslation();

  const { device, onDisconnect } = props;

  return (
    <>
      <Popover
        content={getDeviceConnectedPopoverContent(t, device, onDisconnect)}
        position="bottom"
        inheritDarkTheme={false}
        transitionDuration={isWithReactRevealAnimations() ? 700 : 0}
      >
        <Tooltip
          content={<Text>{t('Click to see more')}</Text>}
          position="right"
          hoverOpenDelay={400}
        >
          <Button
            id="connected-device-info-stepper-button"
            intent="success"
            style={{
              width: '150px',
              height: '10px !important',
              borderRadius: '100px',
              position: 'relative',
              margin: '0 auto',
            }}
          >
            <Row>
              <Col xs={1}>
                <Icon icon="info-sign" />
              </Col>
              <Col xs>
                <Text>{t('Connected')}</Text>
              </Col>
            </Row>
          </Button>
        </Tooltip>
      </Popover>
    </>
  );
}
