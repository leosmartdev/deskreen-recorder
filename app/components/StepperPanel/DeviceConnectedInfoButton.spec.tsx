import React from 'react';
import Enzyme, { mount } from 'enzyme';
import EnzymeToJson from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';
import { BrowserRouter as Router } from 'react-router-dom';
import DeviceConnectedInfoButton from './DeviceConnectedInfoButton';

Enzyme.configure({ adapter: new Adapter() });
jest.useFakeTimers();

it('should match exact snapshot', () => {
  const subject = mount(
    <>
      <Router>
        <DeviceConnectedInfoButton
          device={{} as Device}
          onDisconnect={() => {}}
        />
      </Router>
    </>
  );
  expect(EnzymeToJson(subject)).toMatchSnapshot();
});
