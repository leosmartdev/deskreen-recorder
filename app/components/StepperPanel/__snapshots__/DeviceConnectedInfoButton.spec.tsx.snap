// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match exact snapshot 1`] = `
<BrowserRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <DeviceConnectedInfoButton
      device={Object {}}
      onDisconnect={[Function]}
    >
      <Blueprint3.Popover
        boundary="scrollParent"
        captureDismiss={false}
        content={
          <Row>
            <div
              style={
                Object {
                  "borderRadius": "100px",
                  "padding": "20px",
                }
              }
            >
              <Row
                style={
                  Object {
                    "margin": "0 px 10px 10px 10px",
                  }
                }
              >
                <DeviceInfoCallout />
              </Row>
              <Row>
                <Col
                  xs={12}
                >
                  <Blueprint3.Button
                    icon="disable"
                    intent="danger"
                    onClick={[Function]}
                    style={
                      Object {
                        "borderRadius": "100px",
                        "width": "100%",
                      }
                    }
                  >
                    
                  </Blueprint3.Button>
                </Col>
              </Row>
            </div>
          </Row>
        }
        defaultIsOpen={false}
        disabled={false}
        fill={false}
        hasBackdrop={false}
        hoverCloseDelay={300}
        hoverOpenDelay={150}
        inheritDarkTheme={false}
        interactionKind="click"
        minimal={false}
        modifiers={Object {}}
        openOnTargetFocus={true}
        position="bottom"
        targetTagName="span"
        transitionDuration={0}
        usePortal={true}
        wrapperTagName="span"
      >
        <Manager>
          <span
            className="bp3-popover-wrapper"
          >
            <Reference
              innerRef={[Function]}
            >
              <InnerReference
                innerRef={[Function]}
                setReferenceNode={[Function]}
              >
                <Blueprint3.ResizeSensor
                  onResize={[Function]}
                >
                  <span
                    className="bp3-popover-target"
                    onClick={[Function]}
                  >
                    <Blueprint3.Tooltip
                      className=""
                      content={
                        <Blueprint3.Text
                          ellipsize={false}
                          tagName="div"
                        >
                          
                        </Blueprint3.Text>
                      }
                      hoverCloseDelay={0}
                      hoverOpenDelay={400}
                      key=".0"
                      minimal={false}
                      position="right"
                      transitionDuration={100}
                    >
                      <Blueprint3.Popover
                        autoFocus={false}
                        boundary="scrollParent"
                        canEscapeKeyClose={false}
                        captureDismiss={false}
                        className=""
                        content={
                          <Blueprint3.Text
                            ellipsize={false}
                            tagName="div"
                          >
                            
                          </Blueprint3.Text>
                        }
                        defaultIsOpen={false}
                        disabled={false}
                        enforceFocus={false}
                        fill={false}
                        hasBackdrop={false}
                        hoverCloseDelay={0}
                        hoverOpenDelay={400}
                        inheritDarkTheme={true}
                        interactionKind="hover-target"
                        lazy={true}
                        minimal={false}
                        modifiers={
                          Object {
                            "arrow": Object {
                              "enabled": true,
                            },
                          }
                        }
                        openOnTargetFocus={true}
                        popoverClassName="bp3-tooltip"
                        position="right"
                        targetTagName="span"
                        transitionDuration={100}
                        usePortal={true}
                        wrapperTagName="span"
                      >
                        <Manager>
                          <span
                            className="bp3-popover-wrapper"
                          >
                            <Reference
                              innerRef={[Function]}
                            >
                              <InnerReference
                                innerRef={[Function]}
                                setReferenceNode={[Function]}
                              >
                                <Blueprint3.ResizeSensor
                                  onResize={[Function]}
                                >
                                  <span
                                    className="bp3-popover-target"
                                    onBlur={[Function]}
                                    onFocus={[Function]}
                                    onMouseEnter={[Function]}
                                    onMouseLeave={[Function]}
                                  >
                                    <Blueprint3.Button
                                      className=""
                                      id="connected-device-info-stepper-button"
                                      intent="success"
                                      key=".0"
                                      style={
                                        Object {
                                          "borderRadius": "100px",
                                          "height": "10px !important",
                                          "margin": "0 auto",
                                          "position": "relative",
                                          "width": "150px",
                                        }
                                      }
                                      tabIndex={0}
                                    >
                                      <button
                                        className="bp3-button bp3-intent-success"
                                        id="connected-device-info-stepper-button"
                                        onBlur={[Function]}
                                        onKeyDown={[Function]}
                                        onKeyUp={[Function]}
                                        style={
                                          Object {
                                            "borderRadius": "100px",
                                            "height": "10px !important",
                                            "margin": "0 auto",
                                            "position": "relative",
                                            "width": "150px",
                                          }
                                        }
                                        tabIndex={0}
                                        type="button"
                                      >
                                        <Blueprint3.Icon
                                          key="leftIcon"
                                        />
                                        <span
                                          className="bp3-button-text"
                                          key="text"
                                        >
                                          <Row>
                                            <div
                                              className="row"
                                            >
                                              <Col
                                                xs={1}
                                              >
                                                <div
                                                  className="col-xs-1"
                                                >
                                                  <Blueprint3.Icon
                                                    icon="info-sign"
                                                  >
                                                    <span
                                                      className="bp3-icon bp3-icon-info-sign"
                                                      icon="info-sign"
                                                    >
                                                      <svg
                                                        data-icon="info-sign"
                                                        height={16}
                                                        viewBox="0 0 16 16"
                                                        width={16}
                                                      >
                                                        <desc>
                                                          info-sign
                                                        </desc>
                                                        <path
                                                          d="M8 0C3.58 0 0 3.58 0 8s3.58 8 8 8 8-3.58 8-8-3.58-8-8-8zM7 3h2v2H7V3zm3 10H6v-1h1V7H6V6h3v6h1v1z"
                                                          fillRule="evenodd"
                                                          key="0"
                                                        />
                                                      </svg>
                                                    </span>
                                                  </Blueprint3.Icon>
                                                </div>
                                              </Col>
                                              <Col
                                                xs={true}
                                              >
                                                <div
                                                  className="col-xs"
                                                >
                                                  <Blueprint3.Text
                                                    ellipsize={false}
                                                    tagName="div"
                                                  >
                                                    <div
                                                      className=""
                                                    />
                                                  </Blueprint3.Text>
                                                </div>
                                              </Col>
                                            </div>
                                          </Row>
                                        </span>
                                        <Blueprint3.Icon
                                          key="rightIcon"
                                        />
                                      </button>
                                    </Blueprint3.Button>
                                  </span>
                                </Blueprint3.ResizeSensor>
                              </InnerReference>
                            </Reference>
                            <Blueprint3.Overlay
                              autoFocus={false}
                              backdropClassName="bp3-popover-backdrop"
                              backdropProps={Object {}}
                              canEscapeKeyClose={false}
                              canOutsideClickClose={false}
                              enforceFocus={false}
                              hasBackdrop={false}
                              isOpen={false}
                              lazy={true}
                              onClose={[Function]}
                              transitionDuration={100}
                              transitionName="bp3-popover"
                              usePortal={true}
                            />
                          </span>
                        </Manager>
                      </Blueprint3.Popover>
                    </Blueprint3.Tooltip>
                  </span>
                </Blueprint3.ResizeSensor>
              </InnerReference>
            </Reference>
            <Blueprint3.Overlay
              autoFocus={true}
              backdropClassName="bp3-popover-backdrop"
              backdropProps={Object {}}
              canEscapeKeyClose={true}
              canOutsideClickClose={true}
              enforceFocus={true}
              hasBackdrop={false}
              isOpen={false}
              lazy={true}
              onClose={[Function]}
              transitionDuration={0}
              transitionName="bp3-popover"
              usePortal={true}
            />
          </span>
        </Manager>
      </Blueprint3.Popover>
    </DeviceConnectedInfoButton>
  </Router>
</BrowserRouter>
`;
